presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -401,10 +416,18 @@ public TaskInfo updateTask(
                 queryContext.setMemoryLimits(
                         getQueryMaxMemoryPerNode(session),
                         getQueryMaxTotalMemoryPerNode(session),
-                        getQueryMaxBroadcastMemory(session));
+                        getQueryMaxBroadcastMemory(session),
+                        getQueryMaxRevocableMemoryPerNode(session));
             }
         }
 
+        queryContext.setVerboseExceededMemoryLimitErrorsEnabled(isVerboseExceededMemoryLimitErrorsEnabled(session));
+        queryContext.setHeapDumpOnExceededMemoryLimitEnabled(isHeapDumpOnExceededMemoryLimitEnabled(session));
+        String heapDumpFilePath = Paths.get(
+                getHeapDumpFileDirectory(session),
+                format("%s_%s.hprof", session.getQueryId().getId(), taskId.getStageExecutionId().getStageId().getId())).toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getId&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 428. Consider formatting this data with &#39;%d&#39; instead.